cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/service/http/BasicSystemHttpServiceContext.java

```diff
@@ -63,6 +78,12 @@
   private final NamespaceId namespaceId;
   private final TransactionRunner transactionRunner;
   private final PreferencesFetcher preferencesFetcher;
+  private final RemoteClient remoteTaskClient;
+  private static final Gson GSON = new Gson();
+  private static final int RETRY_LIMIT = 10;
```

Comment:
Retry settings should come from cConf. See how other retry strategies are being configured. You probably can use the `service.retry.policy` policy.

Reply:
Done. Reading from configuration.